erddapy/erddapy.py

```diff
@@ -471,3 +474,18 @@ def get_var_by_attr(self, dataset_id: OptionalStr = None, **kwargs) -> List[str]
             if has_value_flag is True:
                 vs.append(vname)
         return vs
+
+    def download_file(
+        self,
+        file_type,
+    ):
+        """Download the dataset to a file in a user specified format"""
+        url = self.get_download_url(response=file_type)
+        constraints_str = str(dict(sorted(self.constraints.items()))) + str(
+            sorted(self.variables),
+        )
+        constraints_hash = hashlib.sha1(constraints_str.encode("UTF-8")).hexdigest()
```

Comment:
We can probably get by with a smaller hash to avoid huge file names.

```suggestion
        constraints_hash = hashlib.shake_256(constraints_str.encode()).hexdigest(5)
```